toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -0,0 +1,249 @@
+	disks = []Disk{Disk{}}
+	partitionSettings = []PartitionSetting{}
+	curDiskIndex = 0
+	latestDiskIndex = 0
+
+	// Create a mapping of partition flags and corresponding processing functions
+	populatepartCmdProcessMap()
+}
+
+func populatepartCmdProcessMap() {
+	partCmdProcess = make(map[string]func(string) error)
+
+	// Currently only process "ondisk", "ondrive", "size" and "fstype"
+	// Expand this map if supporting more commands in the future
+	partCmdProcess[kickstartPartitionOnDisk] = processDisk
+	partCmdProcess[kickstartPartitionOnDrive] = processDisk
+	partCmdProcess[kickstartPartitionSize] = processPartitionSize
+	partCmdProcess[kickstartPartitionFsType] = processPartitionFsType
+}
+
+func processDisk(InputDiskValue string) (err error) {
+	diskValue := strings.TrimSpace(InputDiskValue)
+	if diskValue == "" {
+		return fmt.Errorf("--ondisk/--ondrive must not be empty")
+	}
+
+	// Check whether this disk has already been parsed or not
+	curIdx, ok := diskInfo[diskValue]
+	if ok {
+		curDiskIndex = curIdx
+	} else {
+		// Don't create new disk if processing on the initial placeholder disk
+		if len(disks) != 1 || len(diskInfo) != 0 {
+			disks = append(disks, Disk{})
+		}
+
+		disks[latestDiskIndex].PartitionTableType = PartitionTableTypeGpt
+
+		// Set TargetDisk and TargetDiskType for unattended installation
+		disks[latestDiskIndex].TargetDisk.Type = "path"
+		disks[latestDiskIndex].TargetDisk.Value = diskValue
+
+		diskInfo[diskValue] = latestDiskIndex
+		curDiskIndex = latestDiskIndex
+		latestDiskIndex++
+	}
+
+	return
+}
+
+func processPartitionSize(InputPartSize string) (err error) {
+	partSize := strings.TrimSpace(InputPartSize)
+
+	if len(disks[curDiskIndex].Partitions) == 0 {
+		newDiskPartition.Start = 1
+	} else {
+		newDiskPartition.Start = disks[curDiskIndex].Partitions[len(disks[curDiskIndex].Partitions)-1].End
+	}
+
+	partitionSize, err := strconv.ParseUint(partSize, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	if !newDiskPartition.Grow {
+		newDiskPartition.End = newDiskPartition.Start + partitionSize
+	}
+
+	return
+}
+
+func processPartitionFsType(inputFsType string) (err error) {
+	fstype := strings.TrimSpace(inputFsType)
+	if fstype == "" {
+		return fmt.Errorf("--fstype must not be empty")
+	}
+
+	if fstype == "" {
+		return fmt.Errorf("fstype cannnot be empty")
+	} else if fstype == "biosboot" {
+		newDiskPartition.FsType = "fat32"
+	} else if fstype == "swap" {
+		newDiskPartition.FsType = "linux-swap"
+
+		// swap partition does not have a mount point
+		newDiskPartitionSetting.MountPoint = ""
+	} else {
+		newDiskPartition.FsType = fstype
+	}
+
+	return
+}
+
+func processMountPoint(InputMountPoint string) (err error) {
+	mountPoint := strings.TrimSpace(InputMountPoint)
+	if mountPoint == "" {
+		return fmt.Errorf("Mount Point must not be empty")
+	}
+
+	newDiskPartitionSetting.MountPoint = mountPoint
+	if mountPoint == "biosboot" {
```

Comment:
These labels are reused in function above as well. They should be declared as consts.